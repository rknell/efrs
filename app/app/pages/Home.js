import React, { Component } from 'react'
import {
  Alert,
  Animated,
  AsyncStorage,
  Button,
  Dimensions,
  Easing,
  Keyboard,
  Platform,
  StatusBar,
  TouchableWithoutFeedback,
  View
} from 'react-native'
import { action, observable } from 'mobx'
import { observer } from 'mobx-react/native'
import TextFieldView from '../componentViews/TextFieldView'
import api from '../api'
import GlobalState from '../services/GlobalState'
import Dashboard from './Dashboard'
import GlobalStyle from '../services/GlobalStyle'
import Domain from '../pages/Domain'
import MyAccount from './MyAccount'
import Analytics from '../services/Analytics'
import Translate from '../services/Translate'

@observer
class HomeView extends Component {
  @observable model = {
    keyboardVisible: false
  }

  measurements = {
    bannerHeight: new Animated.Value(250)
  }

  lastLayouts = {}
  containerSize = {}

  constructor (props) {
    super(props)
    if (props.controller) {
      this.controller = props.controller
    } else {
      this.controller = new HomeController()
    }

    this.keyboardDidShowListener = Keyboard.addListener('keyboardDidShow', (e) => this.loginLayout(e, true))
    this.keyboardDidHideListener = Keyboard.addListener('keyboardDidHide', (e) => this.loginLayout(e, false))
    this.keyboardHeight = 0
  }

  componentWillUnmount () {
    this.keyboardDidShowListener.remove()
    this.keyboardDidHideListener.remove()
  }

  lastAnimateTo = 0

  loginLayout (data, keyboardVisible, type) {
    let loginDimensions = this.lastLayout
    if (keyboardVisible !== undefined) this.model.keyboardVisible = keyboardVisible

    if (type === 'container') {
      this.containerSize = data.nativeEvent.layout
    } else {
      if (data && !data.endCoordinates) {
        this.lastLayouts[type] = data.nativeEvent.layout
        loginDimensions = {
          height: 0,
          width: 0
        }
        Object.keys(this.lastLayouts).forEach(layout => {
          loginDimensions.height += this.lastLayouts[layout].height
          loginDimensions.width += this.lastLayouts[layout].width
        })

        this.lastLayout = loginDimensions
      }

      if (data && data.endCoordinates) {
        this.keyboardHeight = data.endCoordinates.height
      }

      if (keyboardVisible === false) {
        this.keyboardHeight = 0
      }

      if (this.containerSize.height && this.lastLayout.height) {
        let animateTo = this.containerSize.height - this.lastLayout.height - this.keyboardHeight

        if (this.lastAnimateTo !== animateTo) {
          this.lastAnimateTo = animateTo
          Animated.timing(this.measurements.bannerHeight, {
            toValue: animateTo,
            easing: Easing.linear,
            duration: 100
          }).start()
        }
      }
    }
  }

  clicks = 0

  domainClick () {
    this.clicks++

    if (this.clicks === 8) {
      GlobalState.navigation.resetTo({ view: <Domain /> })
    }
  }

  componentDidMount () {
    setTimeout(() => {
      this.loginLayout()
    }, 1)
  }

  render () {
    return (

      <View style={{
        flex: 1,
        backgroundColor: GlobalStyle.style.secondaryColour,
        alignItems: 'center'
      }} onLayout={(data) => this.loginLayout(data, undefined, 'container')}>
        <StatusBar barStyle={GlobalStyle.style.secondaryColourIsDark ? 'light-content' : 'dark-content'} />
        <View onLayout={(data) => this.loginLayout(data, undefined, 'top')}><View style={{ marginTop: 50 }} /></View>
        <Animated.View style={{ height: this.measurements.bannerHeight, alignItems: 'center', justifyContent: 'center' }}>
          <TouchableWithoutFeedback onPress={() => this.domainClick()}>
            <Animated.Image source={GlobalStyle.style.fullLogo} resizeMode='contain' style={{
              height: this.measurements.bannerHeight,
              width: Dimensions.get('window').width * 0.8,
              maxHeight: Dimensions.get('window').height * 0.5
            }} />
          </TouchableWithoutFeedback>
        </Animated.View>

        <View onLayout={(data) => this.loginLayout(data, undefined, 'bottom')}>
          <View style={{
            padding: 16,
            minWidth: Dimensions.get('window').width < 500 ? (Dimensions.get('window').width - 5) : 500
          }}>
            <View style={{
              backgroundColor: GlobalStyle.style.neutralColour,
              borderTopWidth: 4,
              borderColor: GlobalStyle.style.neutralColourHighlight
            }}>
              <TextFieldView
                globalStyle={GlobalStyle.style}
                translate
                label='Username'
                value={this.controller.model.username}
                autoCapitalize='none'
                autoCorrect={false}
                onChange={(text) => {
                  this.controller.updateText('username', text)
                }} />
              <TextFieldView globalStyle={GlobalStyle.style}
                translate
                label='Password'
                autoCapitalize='none'
                value={this.controller.model.password}
                onChange={(text) => {
                  this.controller.updateText('password', text)
                }} secureTextEntry />
            </View>
            <View style={{ marginTop: 30 }}>
              <Button
                color={Platform.OS === 'ios' ? GlobalStyle.style.secondaryColourText : GlobalStyle.style.primaryColour}
                title={Translate.do('Login')}
                onPress={this.controller.login} />
            </View>

          </View>
        </View>

      </View>
    )
  }
}

export class HomeController {
  @observable model = {
    username: '',
    password: '',
    domain: ''
  }

  passwordFailCount = 0

  @action
  updateText (field, value) {
    this.model[field] = value
  }

  constructor () {
    setTimeout(() => {
      GlobalState.model.user = {}
      GlobalState.model.userToken = null
      GlobalState.model.userPermissions = { permissions: {} }
      AsyncStorage.removeItem('@PASettings:authToken')
    }, 1)

    AsyncStorage.getItem(`@login:domain`)
      .then(domain => {
        if (!domain) {
        } else {
          this.model.domain = domain
        }
      })
  }

  @action
  login = () => {
    AsyncStorage.setItem(`@login:domain`, this.model.domain)
    if (this.model.username && this.model.username.trim && !!this.model.username.trim()) {
      this.model.username.trim()
      api.user.login(this.model)
        .then(() => {
          Analytics.logEvent('login_success', { user: this.model.username, domain: this.model.domain })
          if (!GlobalState.model.user.isSetup || GlobalState.model.user.isTempPassword) {
            GlobalState.navigation.resetTo({ view: <MyAccount /> })
          } else {
            GlobalState.navigation.resetTo({ view: <Dashboard /> })
          }
        })
        .catch((err) => {
          console.log(err)
          Analytics.logEvent('login_fail', { user: this.model.username, domain: this.model.domain })
          if (!err.status && (err.body && err.body.message)) {
            Alert.alert(Translate.do('Error logging in'), Translate.do(`There was an unknown error logging you in. Please try again or contact your system administrator`))
          } else if (err.body && err.body.message) {
            if (err.body.type === 'INVALID_PASSWORD') {
              this.passwordFailCount++
              if (this.passwordFailCount > 2) {
                Alert.alert(Translate.do('Password Failed'), Translate.do(`Would you like to reset your password?`), [{
                  text: Translate.do('Try again'),
                  style: 'cancel',
                  onPress: () => {
                    GlobalState.navigation.pop()
                  }
                }, {
                  text: Translate.do('Reset Password'),
                  onPress: () => {
                    api.user.resetPassword(this.model.username)
                      .then(() => {
                        Alert.alert(Translate.do('Password reset'), Translate.do('A new password has been sent to your email.'))
                      })
                  }
                }])
              } else {
                Alert.alert(Translate.do('Error logging in'), Translate.do(err.body.message))
              }
            } else {
              Alert.alert(Translate.do('Error logging in'), Translate.do(err.body.message))
            }
          }
        })
    } else {
      Alert.alert('Login failed', 'Please enter a username')
    }
  }
}

export default HomeView
