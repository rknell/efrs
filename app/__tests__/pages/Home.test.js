import 'react-native'
import React from 'react'
import Home, { HomeController } from '../../app/pages/Home'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'
jest.mock('mobx-react/native', () => require('mobx-react/custom'))

it(`hasn't changed`, () => {
  const tree = renderer.create(<Home />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('has bindings for login, password and domain', () => {
  const homeController = new HomeController()
  homeController.model.password = 'test'
  homeController.model.username = 'test'
  homeController.model.domain = 'test'
  const tree = renderer.create(<Home controller={homeController} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('should update text', () => {
  const homeController = new HomeController()
  homeController.updateText('test', true)
  expect(homeController.model.test).toBe(true)
})
