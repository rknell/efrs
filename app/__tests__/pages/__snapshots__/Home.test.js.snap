// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has bindings for login, password and domain 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f3f3f3",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Object {
          "marginTop": 50,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 250,
        "justifyContent": "center",
      }
    }
  >
    <Image
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../app/assets/efrsFullBanner.png",
        }
      }
      style={
        Object {
          "height": 250,
          "maxHeight": 667,
          "width": 600,
        }
      }
    />
  </View>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Object {
          "minWidth": 500,
          "padding": 16,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "rgb(217, 217, 217)",
            "borderTopWidth": 4,
          }
        }
      >
        <View
          style={
            Object {
              "height": undefined,
              "overflow": "hidden",
              "width": undefined,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "borderBottomWidth": 1,
                "borderColor": "rgb(217, 217, 217)",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "stretch",
                  "color": "rgb(51, 51, 51)",
                  "flex": 1,
                  "fontWeight": "bold",
                  "marginRight": 10,
                  "padding": 10,
                  "paddingLeft": 4,
                  "paddingTop": 10,
                  "textAlign": "right",
                }
              }
            >
              Username
            </Text>
            <View
              style={
                Object {
                  "flex": 2,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderColor": "rgb(217, 217, 217)",
                    "borderWidth": 1,
                    "flex": 1,
                    "flexDirection": "row",
                    "padding": 3,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCorrect={false}
                  autoGrow={true}
                  clearButtonMode="while-editing"
                  editable={true}
                  globalStyle={
                    Object {
                      "disabledText": "rgb(179, 179, 179)",
                      "errorColour": "red",
                      "fullLogo": Object {
                        "testUri": "../../../app/assets/efrsFullBanner.png",
                      },
                      "neutralColour": "white",
                      "neutralColourDisabled": "rgba(255, 255, 255, 0.3)",
                      "neutralColourHighlight": "rgb(217, 217, 217)",
                      "neutralColourIsDark": false,
                      "neutralColourOffset": "rgb(242, 242, 242)",
                      "neutralColourText": "rgb(51, 51, 51)",
                      "neutralColourTextDisabled": "rgba(51, 51, 51, 0.3)",
                      "primaryColour": "dodgerblue",
                      "primaryColourDisabled": "rgba(30, 144, 255, 0.3)",
                      "primaryColourHighlight": "rgb(106, 182, 255)",
                      "primaryColourIsDark": true,
                      "primaryColourText": "rgb(255, 255, 255)",
                      "primaryColourTextDisabled": "rgba(255, 255, 255, 0.3)",
                      "secondaryColour": "#f3f3f3",
                      "secondaryColourDisabled": "rgba(243, 243, 243, 0.3)",
                      "secondaryColourHighlight": "rgb(205, 205, 205)",
                      "secondaryColourIsDark": undefined,
                      "secondaryColourText": "rgb(39, 39, 39)",
                      "secondaryColourTextDisabled": "rgba(39, 39, 39, 0.3)",
                      "secondaryLogo": Object {
                        "testUri": "../../../app/assets/icon.png",
                      },
                    }
                  }
                  keyboardType="ascii-capable"
                  label="Username"
                  onChange={[Function]}
                  onChangeText={[Function]}
                  onContentSizeChange={[Function]}
                  rejectResponderTermination={true}
                  selectionColor="dodgerblue"
                  style={
                    Array [
                      Object {
                        "alignSelf": "stretch",
                        "color": "rgb(51, 51, 51)",
                        "flex": 1,
                        "fontSize": 16,
                        "height": undefined,
                        "padding": 0,
                        "textAlignVertical": "top",
                      },
                    ]
                  }
                  translate={true}
                  underlineColorAndroid="transparent"
                  value="test"
                />
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "height": undefined,
              "overflow": "hidden",
              "width": undefined,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "borderBottomWidth": 1,
                "borderColor": "rgb(217, 217, 217)",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "stretch",
                  "color": "rgb(51, 51, 51)",
                  "flex": 1,
                  "fontWeight": "bold",
                  "marginRight": 10,
                  "padding": 10,
                  "paddingLeft": 4,
                  "paddingTop": 10,
                  "textAlign": "right",
                }
              }
            >
              Password
            </Text>
            <View
              style={
                Object {
                  "flex": 2,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderColor": "rgb(217, 217, 217)",
                    "borderWidth": 1,
                    "flex": 1,
                    "flexDirection": "row",
                    "padding": 3,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoGrow={true}
                  clearButtonMode="while-editing"
                  editable={true}
                  globalStyle={
                    Object {
                      "disabledText": "rgb(179, 179, 179)",
                      "errorColour": "red",
                      "fullLogo": Object {
                        "testUri": "../../../app/assets/efrsFullBanner.png",
                      },
                      "neutralColour": "white",
                      "neutralColourDisabled": "rgba(255, 255, 255, 0.3)",
                      "neutralColourHighlight": "rgb(217, 217, 217)",
                      "neutralColourIsDark": false,
                      "neutralColourOffset": "rgb(242, 242, 242)",
                      "neutralColourText": "rgb(51, 51, 51)",
                      "neutralColourTextDisabled": "rgba(51, 51, 51, 0.3)",
                      "primaryColour": "dodgerblue",
                      "primaryColourDisabled": "rgba(30, 144, 255, 0.3)",
                      "primaryColourHighlight": "rgb(106, 182, 255)",
                      "primaryColourIsDark": true,
                      "primaryColourText": "rgb(255, 255, 255)",
                      "primaryColourTextDisabled": "rgba(255, 255, 255, 0.3)",
                      "secondaryColour": "#f3f3f3",
                      "secondaryColourDisabled": "rgba(243, 243, 243, 0.3)",
                      "secondaryColourHighlight": "rgb(205, 205, 205)",
                      "secondaryColourIsDark": undefined,
                      "secondaryColourText": "rgb(39, 39, 39)",
                      "secondaryColourTextDisabled": "rgba(39, 39, 39, 0.3)",
                      "secondaryLogo": Object {
                        "testUri": "../../../app/assets/icon.png",
                      },
                    }
                  }
                  keyboardType="ascii-capable"
                  label="Password"
                  onChange={[Function]}
                  onChangeText={[Function]}
                  onContentSizeChange={[Function]}
                  rejectResponderTermination={true}
                  secureTextEntry={true}
                  selectionColor="dodgerblue"
                  style={
                    Array [
                      Object {
                        "alignSelf": "stretch",
                        "color": "rgb(51, 51, 51)",
                        "flex": 1,
                        "fontSize": 16,
                        "height": undefined,
                        "padding": 0,
                        "textAlignVertical": "top",
                      },
                    ]
                  }
                  translate={true}
                  underlineColorAndroid="transparent"
                  value="test"
                />
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 30,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "rgb(39, 39, 39)",
                  },
                ]
              }
            >
              Login
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`hasn't changed 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f3f3f3",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Object {
          "marginTop": 50,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 250,
        "justifyContent": "center",
      }
    }
  >
    <Image
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../app/assets/efrsFullBanner.png",
        }
      }
      style={
        Object {
          "height": 250,
          "maxHeight": 667,
          "width": 600,
        }
      }
    />
  </View>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Object {
          "minWidth": 500,
          "padding": 16,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "rgb(217, 217, 217)",
            "borderTopWidth": 4,
          }
        }
      >
        <View
          style={
            Object {
              "height": undefined,
              "overflow": "hidden",
              "width": undefined,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "borderBottomWidth": 1,
                "borderColor": "rgb(217, 217, 217)",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "stretch",
                  "color": "rgb(51, 51, 51)",
                  "flex": 1,
                  "fontWeight": "bold",
                  "marginRight": 10,
                  "padding": 10,
                  "paddingLeft": 4,
                  "paddingTop": 10,
                  "textAlign": "right",
                }
              }
            >
              Username
            </Text>
            <View
              style={
                Object {
                  "flex": 2,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderColor": "rgb(217, 217, 217)",
                    "borderWidth": 1,
                    "flex": 1,
                    "flexDirection": "row",
                    "padding": 3,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCorrect={false}
                  autoGrow={true}
                  clearButtonMode="while-editing"
                  editable={true}
                  globalStyle={
                    Object {
                      "disabledText": "rgb(179, 179, 179)",
                      "errorColour": "red",
                      "fullLogo": Object {
                        "testUri": "../../../app/assets/efrsFullBanner.png",
                      },
                      "neutralColour": "white",
                      "neutralColourDisabled": "rgba(255, 255, 255, 0.3)",
                      "neutralColourHighlight": "rgb(217, 217, 217)",
                      "neutralColourIsDark": false,
                      "neutralColourOffset": "rgb(242, 242, 242)",
                      "neutralColourText": "rgb(51, 51, 51)",
                      "neutralColourTextDisabled": "rgba(51, 51, 51, 0.3)",
                      "primaryColour": "dodgerblue",
                      "primaryColourDisabled": "rgba(30, 144, 255, 0.3)",
                      "primaryColourHighlight": "rgb(106, 182, 255)",
                      "primaryColourIsDark": true,
                      "primaryColourText": "rgb(255, 255, 255)",
                      "primaryColourTextDisabled": "rgba(255, 255, 255, 0.3)",
                      "secondaryColour": "#f3f3f3",
                      "secondaryColourDisabled": "rgba(243, 243, 243, 0.3)",
                      "secondaryColourHighlight": "rgb(205, 205, 205)",
                      "secondaryColourIsDark": undefined,
                      "secondaryColourText": "rgb(39, 39, 39)",
                      "secondaryColourTextDisabled": "rgba(39, 39, 39, 0.3)",
                      "secondaryLogo": Object {
                        "testUri": "../../../app/assets/icon.png",
                      },
                    }
                  }
                  keyboardType="ascii-capable"
                  label="Username"
                  onChange={[Function]}
                  onChangeText={[Function]}
                  onContentSizeChange={[Function]}
                  rejectResponderTermination={true}
                  selectionColor="dodgerblue"
                  style={
                    Array [
                      Object {
                        "alignSelf": "stretch",
                        "color": "rgb(51, 51, 51)",
                        "flex": 1,
                        "fontSize": 16,
                        "height": undefined,
                        "padding": 0,
                        "textAlignVertical": "top",
                      },
                    ]
                  }
                  translate={true}
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "height": undefined,
              "overflow": "hidden",
              "width": undefined,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "borderBottomWidth": 1,
                "borderColor": "rgb(217, 217, 217)",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "stretch",
                  "color": "rgb(51, 51, 51)",
                  "flex": 1,
                  "fontWeight": "bold",
                  "marginRight": 10,
                  "padding": 10,
                  "paddingLeft": 4,
                  "paddingTop": 10,
                  "textAlign": "right",
                }
              }
            >
              Password
            </Text>
            <View
              style={
                Object {
                  "flex": 2,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderColor": "rgb(217, 217, 217)",
                    "borderWidth": 1,
                    "flex": 1,
                    "flexDirection": "row",
                    "padding": 3,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoGrow={true}
                  clearButtonMode="while-editing"
                  editable={true}
                  globalStyle={
                    Object {
                      "disabledText": "rgb(179, 179, 179)",
                      "errorColour": "red",
                      "fullLogo": Object {
                        "testUri": "../../../app/assets/efrsFullBanner.png",
                      },
                      "neutralColour": "white",
                      "neutralColourDisabled": "rgba(255, 255, 255, 0.3)",
                      "neutralColourHighlight": "rgb(217, 217, 217)",
                      "neutralColourIsDark": false,
                      "neutralColourOffset": "rgb(242, 242, 242)",
                      "neutralColourText": "rgb(51, 51, 51)",
                      "neutralColourTextDisabled": "rgba(51, 51, 51, 0.3)",
                      "primaryColour": "dodgerblue",
                      "primaryColourDisabled": "rgba(30, 144, 255, 0.3)",
                      "primaryColourHighlight": "rgb(106, 182, 255)",
                      "primaryColourIsDark": true,
                      "primaryColourText": "rgb(255, 255, 255)",
                      "primaryColourTextDisabled": "rgba(255, 255, 255, 0.3)",
                      "secondaryColour": "#f3f3f3",
                      "secondaryColourDisabled": "rgba(243, 243, 243, 0.3)",
                      "secondaryColourHighlight": "rgb(205, 205, 205)",
                      "secondaryColourIsDark": undefined,
                      "secondaryColourText": "rgb(39, 39, 39)",
                      "secondaryColourTextDisabled": "rgba(39, 39, 39, 0.3)",
                      "secondaryLogo": Object {
                        "testUri": "../../../app/assets/icon.png",
                      },
                    }
                  }
                  keyboardType="ascii-capable"
                  label="Password"
                  onChange={[Function]}
                  onChangeText={[Function]}
                  onContentSizeChange={[Function]}
                  rejectResponderTermination={true}
                  secureTextEntry={true}
                  selectionColor="dodgerblue"
                  style={
                    Array [
                      Object {
                        "alignSelf": "stretch",
                        "color": "rgb(51, 51, 51)",
                        "flex": 1,
                        "fontSize": 16,
                        "height": undefined,
                        "padding": 0,
                        "textAlignVertical": "top",
                      },
                    ]
                  }
                  translate={true}
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 30,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "rgb(39, 39, 39)",
                  },
                ]
              }
            >
              Login
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
